leaktest.go

```diff
@@ -13,42 +13,74 @@ import (
 	"context"
 	"runtime"
 	"sort"
+	"strconv"
 	"strings"
 	"time"
 )
 
+type goroutine struct {
+	id    uint64
+	stack string
+}
+
+type goroutineByID []goroutine
+
+func (g goroutineByID) Len() int           { return len(g) }
+func (g goroutineByID) Less(i, j int) bool { return g[i].id < g[j].id }
+func (g goroutineByID) Swap(i, j int)      { g[i], g[j] = g[j], g[i] }
+
+func interestingGoroutine(g string) (gr goroutine, ok bool) {
```

Comment:
No problem. Would you prefer to have the case where there's no error but the goroutine is not interesting return nil and an error, or return nil, nil?